gabarit/template_num/num_project/package_name/monitoring/mlflow_logger.py

```diff
@@ -17,134 +17,52 @@
-        sock.shutdown(socket.SHUT_RDWR)
-        reachable = True
-    except Exception:
-        logger.error(f'Monitoring - MlFlow  @ {host} not reachable => nothing will be stored')
-    finally:
-        sock.close()
-
-    # Return state
-    return reachable
-
-
-def is_local(host: str) -> bool:
-    '''Checks if mlflow is running in local
-
-    Args:
-        host (str): URI of the host
-    Returns:
-        bool: If mlflow is running in local
-    '''
-    l1 = len(host)
-    host = re.sub(r'(?i)http(s)*://', '', host)
-    l2 = len(host)
-    if l1 == l2:  # no http
-        return True
-    else:
-        return False
-
-
-def is_mlflow_up(func: Callable) -> Callable:
-    '''Checks if mlflow server is up & running before calling the decorated function
-
-    Args:
-        func (Callable): Function to decorate
-    Returns:
-        Callable: Wrapper
-    '''
-
-    # Define wrapper to check if mlflow is up
-    def wrapper(self, *args, **kwargs):
-
-        # We run only if running == True (ie. initial connection ok)
-        if self.running:
-
-            # Check if we can run
-            if is_local(self.tracking_uri):
-                to_run = True  # OK because local
-            elif is_running(self.tracking_uri, 80, self.logger):
-                to_run = True  # OK because still running
-            else:
-                to_run = False  # KO
-
-            # run if possible
-            if to_run:
-                try:
-                    func(self, *args, **kwargs)
-                except Exception as e:  # Manage mlflow errors (but continues process)
-                    self.logger.error("Can't log on mlflow")
-                    self.logger.error(repr(e))
-            # Else : do nothing (error already logged)
-
-    # For test purposes (ignore wrapper)
-    wrapper.wrapped_fn = func  # type: ignore
-
-    return wrapper
-
-
-class ModelLogger:
+class MLflowLogger:
     '''Abstracts how MlFlow works'''
 
-    _default_name = f'{{package_name}}-approche-{uuid.uuid4()}'
-    _default_tracking_uri = ''
-
-    def __init__(self, tracking_uri: Union[str, None] = None, experiment_name: Union[str, None] = None) -> None:
+    def __init__(self, experiment_name: str, tracking_uri: str = '') -> None:
         '''Class initialization
-
+        Args:
+            experiment_name (str):  Name of the experiment to activate
         Kwargs:
             tracking_uri (str): URI of the tracking server
-            experiment_name (str): Name of the experiment to activate
         '''
         # Get logger
         self.logger = logging.getLogger(__name__)
+
         # Set tracking URI & experiment name
-        self.tracking_uri = tracking_uri if tracking_uri is not None else self._default_tracking_uri
-        self.experiment_name = experiment_name if experiment_name is not None else self._default_name
-        # Initiate tracking
-        # There is a try...except in order to test if mlflow is reachable
-        try:
```

Comment:
Shouldn't we keep a running argument if can't reach the host ? Or at least backup on local saves ?